src/symbols_linux.cpp

```diff
@@ -457,7 +461,56 @@ void ElfParser::loadSymbols(bool use_debug) {
     }
 }
 
-// Load symbols from /usr/lib/debug/.build-id/ab/cdef1234.debug, where abcdef1234 is Build ID
+const char *ElfParser::getDebuginfodCache() {
+    if(_debuginfod_cache_buf[0]) {
+        return _debuginfod_cache_buf;
+    }
+
+    const char* env_vars[] = {"DEBUGINFOD_CACHE_PATH", "XDG_CACHE_HOME", "HOME"};
+    const char* suffixes[] = {"", "debuginfod_client/", ".cache/debuginfod_client/"};
+
+    for (int i = 0; i < sizeof(env_vars); ++i) {
+        const char* env_val = getenv(env_vars[i]);
+        if (!env_val || strlen(env_val) == 0) {
```

Comment:
`env_val[0]` is a better way to check for non-empty string, you already do this in other places.